src/fireedge/src/modules/components/Forms/VrTemplate/InstantiateForm/Steps/UserInputs/Node/UserInputDialog.js

```diff
@@ -0,0 +1,176 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2025, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+import { useEffect } from 'react'
+import { Component, useFormContext, Controller } from 'react-hook-form'
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  DialogActions,
+  Button,
+  Typography,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  FormControlLabel,
+  RadioGroup,
+  Radio,
+} from '@mui/material'
+import PropTypes from 'prop-types'
+import { Tr } from '@modules/components/HOC'
+import { T } from '@ConstantsModule'
+
+/**
+ * @param {object} root0 - Props
+ * @param {boolean} root0.open - Is dialog open?
+ * @param {Function} root0.onClose - On close handler
+ * @param {object} root0.userInput - User input
+ * @returns {Component} - User input edit dialog
+ */
+const UserInputDialog = ({ open, onClose, userInput }) => {
+  const { control, setValue } = useFormContext({
+    defaultValues: {
+      value: userInput?.default || undefined,
+    },
+  })
+
+  useEffect(() => {
+    setValue(`user_inputs.${userInput.name}`, userInput?.default || '')
+  }, [userInput, setValue])
+
+  const onSubmit = () => {
+    onClose()
+  }
+
+  const renderInputField = ({ field }) => {
+    const handleNumericChange = (event) => {
+      const value = event.target.value
+      field.onChange(userInput.type === 'number' ? Number(value) : value)
+    }
+    switch (userInput.type) {
+      case 'list':
+      case 'listmultiple':
+        return (
+          <FormControl fullWidth>
+            <InputLabel id="value-select-label">Value</InputLabel>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX element content is not internationalized. Application should support multiple languages through internationalization. Instead, utilize packages like &#39;i18next&#39; to internationalize your items.